dvc/dependency/repo.py

```diff
@@ -71,7 +72,15 @@ def fetch(self):
 
             out = repo.find_out_by_relpath(self.def_path)
             with repo.state:
-                repo.cloud.pull(out.get_used_cache())
+                try:
+                    repo.cloud.pull(out.get_used_cache())
+                except NoRemoteError:
+                    # Check if the file is available in the cache
```

Comment:
The comment is describing _what_ the code is doing, but I would appreciate more a comment about _why_ it is doing such thing.

Reply:
I have updated the comment. Let me know if this looks ok?